src/sankey_diagram_plotly.py

```diff
@@ -350,43 +251,53 @@ def add_node(node_name):
         "primary_runner",
     ] = "No test runner detected"
 
-    # For each runner, calculate the number of repositories that primarily use it
+    # For each runner, calculate the number of repositories that primarily use
+    # it
     logger.info(
-        "For each runner, calculate the number of repositories that " "primarily use it"
+        "For each runner, calculate the number of repositories that primarily" " use it"
```

Comment:
very minor tweak: the number that primarily use_s_ it as we're agreeing with the word number.

Reply:
Why are we emphasising "primarily"? what happens when a repo contains one each of two distinct test runners? and how does the code determine which is the primary test running in an individual repo?

Reply:
This was one one of the decisions I had to make regarding the flow of data(number of repos in our case). There are repos which more than one test runners have been detected in them. The script will count the number of file patterns matching each test runner and chooses the one with the most file pattern count as the primary test runner. We can choose another approach where we consider all of the test runners in one repo but then that repo could flow a couple of times into the next layer.

Reply:
Sure. This information is saved in the dataframe `updated_local_github_df_test_count.csv` when the script `src/github_repo_request_local.py` is run.